### PR TITLE
fix: expand tabs by terminal cells, not chars

### LINKED ISSUE #23: expand_tabs counts chars, not terminal cells — tabs after wide/zero-width Unicode land a stop off
Follow-up to the tab-expansion fix in #22, tracking the finding deliberately deferred there (thread: https://github.com/JonasBaeumer/herdr-file-annotator/pull/22#discussion_r3861960573).

**Problem**

`diff::expand_tabs` advances its column counter by one per `char`, not by rendered terminal width. A tab that follows wide (CJK, emoji) or zero-width (combining marks) characters therefore expands to the wrong number of spaces — e.g. `"界\tx"` gets three spaces to "reach" a 4-column stop, but `界` occupies two cells, so only two are needed. Every later character on that line sits one cell off per miscounted glyph.

**Why it was deferred in #22**

The function's own doc scopes it to the ASCII indentation tabs that occur in practice (leading tabs, where nothing precedes them and the count is exact). The miscounting only shows for tabs *after* non-ASCII text — mid-line tabs in CJK comments or emoji-bearing strings — which is cosmetic misalignment, not the render corruption #22 fixes. Landing the corruption fix wasn't worth blocking on the long tail.

**Proposed fix**

Count columns with `unicode_width::UnicodeWidthChar` (already a dependency — it's what the pan/clip math uses, so this also makes `expand_tabs` consistent with the rest of the width handling). Zero-width chars advance 0, wide chars 2. Needs tests for wide-before-tab, zero-width-before-tab, and the existing ASCII cases staying byte-identical.

**Scope note:** display-only; annotations are line-anchored, so no protocol impact.

### PR DESCRIPTION
**Pull request by JonasBaeumer**
Closes #23 — the follow-up deferred from #22's review.

`expand_tabs` counted its column one per `char`, so a tab following wide (CJK, emoji) or zero-width (combining-mark) characters expanded to the wrong number of spaces: `"界\tx"` got three spaces where the terminal needs two, shifting everything after by one cell per miscounted glyph.

The column now advances by `unicode_width::UnicodeWidthChar` — already a dependency, and the exact width model the pane's pan/clip math uses, so tab expansion and the rest of the width handling finally agree. Control characters count 0, matching how the render path treats them. Per-char width summing is also what `UnicodeWidthStr` does for the grapheme clusters downstream, so there's no divergence for ZWJ sequences either.

ASCII behavior is byte-identical (the existing `expand_tabs_aligns_to_tab_stops` test pins it). New regression test covers wide-before-tab, double-wide reaching a full stop, combining-mark, and control-char cases — verified failing against the per-char counting. 141 tests, warning-free.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by JonasBaeumer** [comment]
@codex review

File: src/diff.rs [hunk 1]
```diff
@@ -66,7 +70,7 @@ pub fn expand_tabs(text: &str) -> String {
             col += n;
         } else {
             out.push(ch);
-            col += 1;
+            col += ch.width().unwrap_or(0);
```
[the remark this answers is not in the document]
Comment (JonasBaeumer) [reply src/diff.rs]: Confirmed, not a false alarm: reproduced with your exact example. Copying `expand_tabs` verbatim and running it on `"👩‍👩‍👧‍👦\tx"` (4-codepoint-chain family, 7 chars incl. ZWJs):

- naive per-char sum = 8 (4 emoji chars × width 2, ZWJs × width 0)
- `UnicodeWidthStr::width` on the grapheme cluster (what `ui::str_cols`/`pan_and_clip` actually uses) = 2
- `expand_tabs` output: `"👩‍👩‍👧‍👦" + 4 spaces + "x"` → render column of `x` = 6, vs. the intended tab stop at 4

So per-char summing does *not* match the render path's grapheme-cluster model for multi-scalar sequences — only for single-scalar chars (CJK, combining marks, control chars), which is what this PR's tests cover. Same divergence shows up for VS16 emoji-presentation sequences (e.g. `❤️` = U+2764 U+FE0F: naive=1, grapheme width=2).

Fixing this means switching `expand_tabs` to grapheme-cluster width (pulling `unicode-segmentation` into the tab-expansion path, mirroring `pan_and_clip`), which is a design change beyond this PR's scope (closes #23: char-vs-cell counting for single-scalar chars). Deferring — tracking as a known follow-up rather than fixing here. Leaving this thread open/unresolved for visibility.

[the remark this answers is not in the document]
Comment (JonasBaeumer) [reply src/diff.rs]: Fixed in 9859667 — your repro settled it: the column now advances per grapheme cluster via UnicodeWidthStr (ratatui's own measurement unit), so the family-emoji case gets its 2 cells and VS16 sequences their emoji-presentation width. Regression tests cover both, verified failing against the per-char counting. One side correction fell out: stray control chars now count 1 (string width) rather than 0 (char width) — agreeing with the buffer's accounting is the function's entire job, and the buffer uses string width.

**Comment by JonasBaeumer** [comment]
@codex review
